clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -619,11 +663,12 @@ PointerVariableConstraint::mkString(const EnvironmentMap &E,
   uint32_t TypeIdx = 0;
 
   auto It = vars.begin();
+  auto i = 0;
   // Skip over first pointer level if only emitting pointee string.
   // This is needed when inserting type arguments.
   if (EmitPointee)
     ++It;
-  for (; It != vars.end(); ++It) {
+  for (; It != vars.end() && IMPLIES(std::get<0>(typedeflevelinfo), i < std::get<1>(typedeflevelinfo)); ++It, i++) {
```

Comment:
Add comment to say what this is doing. I wonder: If you are going to define macros like `IMPLIES` maybe also defined a macro for `std::get<0>(X)` as `HAS_TYPEDEF(X)` so it's a little easier to read/understand? Same with the other fields of the tuple.

Reply:
Sure thing

Reply:
Thinking about it, would you prefer a macro or a class with proper method names?

Reply:
Middle ground: You could define a `struct` type rather than a tuple, and then the field names would be more useful.